### DISCUSSION TITLE
deploying static web app resulted in 404 when accessing pages directly from outside the app

### DISCUSSION THREAD

**Discussion by sulhome** [vercel/next.js#41466] [Help] [answered]
I have created a static app using `next export` command and was able to deploy it to vercel just fine.

I am able to navigate through the app with no issues however when I link to a direct url withing the app lets say `/about` then when accessing that URL directly (by clicking on a link) you get redirected to 404 not found. Same behaviour happens if you enter the URL manually.
Thank you for your help.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The issue lays within how you serve your pages.

If you see the `out` folder from `next export` you'll see that there's an `about.html` page, but not an `about` path to follow. 


```
➜ ls out/
404.html
_next
assets
about.html
favicon.ico
index.html
vercel.svg
```

Your server needs to be able to get a request for `/about`, look up `about.html` and serve that. If there's no match, it should serve `404.html`.

In this discussion thread people have posted various server engine configurations, for NGINX, or Apache: https://github.com/vercel/next.js/discussions/10522, you need to find one, or write one that works for your server engine.

## Alternative

You may also use trailing slashes.

In your `next.config.js`:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  trailingSlash: true,
  reactStrictMode: true,
};

module.exports = nextConfig;
```

Now your paths are not `/about` but `/about/`, so you'll have to update your `next/link` and in general anchor tags and navigation on your page.

The `out` folder now contains:

```
404
404.html
_next
assets
about
favicon.ico
index.html
static
vercel.svg
```

Notice that `about` is no longer an HTML file, but a folder, and within it you'll find `index.html`, so when a request arrives for `/about/`, servers respond with `/about/index.html`.

**Reply by icyJoseph** [reply]
@sulhome try this, https://vercel.com/docs/project-configuration#project-configuration/clean-urls

**Reply by sulhome** [reply]
This worked mate. Thank you

**Comment by sulhome** [comment]
Thank you @icyJoseph 
When you say `your server` what do you mean by that? I am using vercel to host the static app so I don't know what server do they use or if I have access to configure that server
